PdCore/build.gradle

```diff
@@ -79,20 +83,20 @@ android {
     }
 }
 
+// TODO: Move to convention plugin?
 def getNdkBuildExecutablePath() {
-    File ndkDir = android.ndkDirectory
-    if (ndkDir == null) {
-        throw new Exception('NDK location not found. Define location with ndk.dir in the ' +
-                'local.properties file or with an ANDROID_NDK_HOME environment variable.')
+    // android.ndkDirectory should return project.android.ndkVersion ndkDirectory
+    def ndkDir = android.ndkDirectory.absolutePath
+    def ndkBuildName = Os.isFamily(Os.FAMILY_WINDOWS) ? 'ndk-build.cmd' : 'ndk-build'
+    def ndkBuildFullPath = new File(ndkDir, ndkBuildName).getAbsolutePath()
+    if (!new File(ndkBuildFullPath).canExecute()) {
+        throw new GradleScriptException("ndk-build executable not found: $ndkBuildFullPath")
     }
-    def isWindows = System.properties['os.name'].toLowerCase().contains('windows')
-    def ndkBuildFile = new File(ndkDir, isWindows ? 'ndk-build.cmd' : 'ndk-build')
-    if (!ndkBuildFile.exists()) {
-        throw new Exception("ndk-build executable not found: $ndkBuildFile.absolutePath")
-    }
-    ndkBuildFile.absolutePath
+    return ndkBuildFullPath
 }
 
+def ndkNBuildExecutablePath = "${{-> getNdkBuildExecutablePath()}.memoize()}"
```

Comment:
Where is `ndkNBuildExecutablePath` used?

Reply:
Used on lines 50 and 68, where the method used to be called eagerly.

Reply:
Lines 50 and 68 refer to `ndkBuildExecutablePath`. This variable is called `ndkNBuildExecutablePath` (notice the 'N' between `ndk` and `BuildExecutablePath`).

Reply:
Wow. That means that the version in the other script must be called above!

I'll fix that.

Reply:
Apparently the reference to `ndkBuildExecutablePath` is  resulting in a call to `getNdkBuildExecutablePath()` but it is also delaying evaluation enough that `android.ndkDirectory` is able to resolve.